sentry_zendesk/plugin.py

```diff
@@ -0,0 +1,221 @@
+            'label': 'Zendesk URL',
+            'default': self.get_option('zendesk_url', project),
+            'type': 'text',
+            'placeholder': 'e.g. "https://mycompany.zendesk.com"',
+            'help': 'It must be visible to the Sentry server'
+        }, {
+            'name': 'username',
+            'label': 'Username',
+            'default': self.get_option('username', project),
+            'type': 'text',
+            'help': 'Ensure the Zendesk user has admin permissions on the '
+                    'project'
+        }, secret_field, {
+            'name': 'auto_create_problems',
+            'label': 'Automatically create Zendesk problems',
+            'default': self.get_option('auto_create_problems', project) or False,  # noqa
+            'type': 'bool',
+            'required': False,
+            'help': 'Automatically create a Zendesk ticket of type problem '
+                    'for EVERY new issue'
+        }, {
+            'name': 'auto_create_incidents',
+            'label': 'Automatically create Zendesk incidents',
+            'default': self.get_option('auto_create_incidents', project) or False,  # noqa
+            'type': 'bool',
+            'required': False,
+            'help': 'Automatically create a Zendesk ticket of type incident ' \
+                    'for EVERY event after the first one, linking it to the ' \
+                    'previously created problem.'
+        }]
+
+    def post_process(self, group, event, is_new, is_sample, **kwargs):
+        """
+        Called by the worker process whenever a new event arrives.
+        """
+        logger.info('event: {}, is_new: {}'.format(event, is_new))
+
+        if is_new:
+            if not self.get_option('auto_create_problems', group.project):
+                return
+            logger.info('New problem')
+            problem_id = self._get_linked_ticket(group)
+            if problem_id:
+                logger.error('There is already a problem linked to this event')
+                return
+
+            logger.info('Creating new problem')
+            ticket_id = self._create_ticket(
+                group, event, ticket_type='problem')
+            GroupMeta.objects.set_value(
+                group, '%s:tid' % self.get_conf_key(), ticket_id)
+        elif self.get_option('auto_create_incidents', group.project):
+            problem_id = self._get_linked_ticket(group)
+            if not problem_id:
+                logger.info(
+                    'Cannot create incident because there is no linked problem'
+                )
+                return
+
+            logger.info(
+                'Creating new incident linked to problem "{}"'
+                .format(problem_id))
+            self._create_ticket(
+                group, event, ticket_type='incident', problem_id=problem_id)
+
+    def _get_linked_ticket(self, group):
+        # XXX(dcramer): Sentry doesn't expect GroupMeta referenced here so we
```

Comment:
Yeah, this is from the official Jira plugin. I preferred to leave the comment as this seems to be something that they may break in the future.